sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosPagedFlux.java

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import com.azure.core.http.rest.PagedFluxBase;
+import reactor.core.publisher.Mono;
+
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/**
+ * This class is a paged flux that can operate on any type that extends {@link FeedResponse} and
+ * also provides the ability to operate on individual items. When processing the response by page,
+ * each response will contain the items in the page as well as the request details like
+ * status code and headers.
+ *
+ * @param <T> The type of items in {@code FeedResponse}.
+ *
+ * @see com.azure.core.http.rest.PagedResponse
+ * @see com.azure.core.http.rest.Page
+ * @see com.azure.core.http.rest.PagedFlux
+ */
+public class CosmosPagedFlux<T> extends PagedFluxBase<T, FeedResponse<T>> {
```

Comment:
This will eagerly compute the continuationToken. I thought CosmosDB wanted to have a lazy evaluation of continuationToken.